pkg/pillar/utils/cipherutils.go

```diff
@@ -19,20 +19,22 @@ import (
 )
 
 // PrepareCipherCred : fill-in the plain-text credentials
-func PrepareCipherCred(id, password string) zconfig.CredentialBlock {
+func PrepareCipherCred(id, password string) (zconfig.CredentialBlock, error) {
 	cred := zconfig.CredentialBlock{}
-	if len(id) != 0 || len(password) != 0 {
-		cred.Identity = id
-		cred.Password = password
+	if len(id) == 0 || len(password) == 0 {
+		return cred, fmt.Errorf("Empty credentials in plain text")
```

Comment:
Does this pass testing? Something like a docker.io download would hit this since it has no datastore credentials.

Reply:
Yes, it is failing. Thanks for pointing it out.